src/checkers/inference/model/serialization/CnfVecIntSerializer.java

```diff
@@ -151,7 +151,10 @@ public CnfVecIntSerializer(SlotManager slotManager) {
         //TODO: WE SHOULD INSTEAD PIPE THROUGH THE ExistentialVariable ID
         Integer existentialId = existentialToPotentialVar.get(constraint.getPotentialVariable().getId());
         if (existentialId == null) {
-            existentialId = slotManager.nextId();
+            // existentialId should not overlap with the Id of real slots in slot manager
+            // thus by computing sum of total slots number in slot manager
+            // and the size of existentialToPotentialVar and plus 1 to get next id of existential Id here
+            existentialId = slotManager.getNumberOfSlots() + existentialToPotentialVar.size() + 1;
```

Comment:
Hi Prof. Dietl, I'm still a bit confused about how we encoding the slots to bit variables in a SAT solver. I'm looking through the code of `CnfVecIntSerializer`, and it seems directly convert slot id to `VecInt` type  without any multiplication in a serialization method, such like this:

```java 

 @Override
            protected VecInt[] variable_variable(VariableSlot slot1, VariableSlot slot2, InequalityConstraint constraint) {

                // a <=> !b which is the same as (!a v !b) & (b v a)
                return new VecInt[]{
                        asVec(-slot1.getId(), -slot2.getId()),
                        asVec( slot1.getId(),  slot2.getId())
                };
            }
//asVec:
 VecInt asVec(int ... vars) {
        return new VecInt(vars);
    }
```

and this:

```java

        for (VecInt clause : potentialClauses) {
            clause.insertFirst(-existentialId);
        }

        for (VecInt clause : alternativeClauses) {
            clause.insertFirst(existentialId);
        }
```
Is the above code means we directly use a slot id as a bit for the SAT solver? (e.g. we directly insert `existentialId` into a clause by calling `clause.insertFirst(existentialId);`)

How to determine wether a slot id is used as a bit for the SAT solver or not?

Thanks!